antarest/study/business/general_management.py

```diff
@@ -0,0 +1,282 @@
+from enum import Enum
+from typing import Optional, Dict, Any, List
+
+from pydantic import StrictStr, StrictBool, conint, PositiveInt
+
+from antarest.study.business.utils import (
+    FormFieldsBaseModel,
+    FieldInfo,
+    GENERAL_DATA_PATH,
+    execute_or_add_commands,
+)
+from antarest.study.model import Study
+from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
+from antarest.study.storage.storage_service import StudyStorageService
+from antarest.study.storage.variantstudy.model.command.update_config import (
+    UpdateConfig,
+)
+from antarest.study.storage.variantstudy.model.command_context import (
+    CommandContext,
+)
+
+
+class Mode(str, Enum):
+    ECONOMY = "Economy"
+    ADEQUACY = "Adequacy"
+    DRAFT = "draft"
+
+
+class Month(str, Enum):
+    JANUARY = "january"
+    FEBRUARY = "february"
+    MARCH = "march"
+    APRIL = "april"
+    MAY = "may"
+    JUNE = "june"
+    JULY = "july"
+    AUGUST = "august"
+    SEPTEMBER = "september"
+    OCTOBER = "october"
+    NOVEMBER = "november"
+    DECEMBER = "december"
+
+
+class WeekDay(str, Enum):
+    MONDAY = "Monday"
+    TUESDAY = "Tuesday"
+    WEDNESDAY = "Wednesday"
+    THURSDAY = "Thursday"
+    FRIDAY = "Friday"
+    SATURDAY = "Saturday"
+    SUNDAY = "Sunday"
+
+
+class BuildingMode(str, Enum):
+    AUTOMATIC = "Automatic"
+    CUSTOM = "Custom"
+    DERATED = "Derated"
+
+
+DayNumberType = conint(ge=1, le=365)
+
+
+class GeneralFormFields(FormFieldsBaseModel):
+    mode: Optional[Mode]
+    first_day: Optional[DayNumberType]  # type: ignore
+    last_day: Optional[DayNumberType]  # type: ignore
+    horizon: Optional[StrictStr]
```

Comment:
let loose with just "str" as sometimes this contains a casted int (like 2030)

Reply:
in general, don't put StrictStr because the FileTree IniReader will always cast a string to a more specific type if possible